src/shared/utils.test.ts

```diff
@@ -0,0 +1,31 @@
+import { roundBinSize } from './utils';
+
+const BIN_SIZES_INTEGER = [1, 2, 3, 5, 10, 15, 20];
+const BIN_SIZES_REAL = [1, 1.25, 2, 2.5, 3, 5, 7.5, 10];
+
+test('roundBinSize', () => {
```

Comment:
Since `utils` offers more than one function, it should `describe('roundBinSize', () => {`
and inside there would be `test('case', () => {` for various situations.

```ts
describe('roundBinSize', () => {
  test('Rounds to itself', () => {
    BIN_SIZES_INTEGER.map((x) => expect(roundBinSize(x, BIN_SIZES_INTEGER)).toBe(x));
    BIN_SIZES_REAL.map((x) => expect(roundBinSize(x, BIN_SIZES_REAL)).toBe(x));
  })

  test('Integer rounding', () => {
    const roundsTo = (in: number, out: number) => expect(roundBinSize(in, BIN_SIZES_INTEGER)).toBe(out))
    roundsTo(1.3, 1);
    roundsTo(1.7, 2);
    ...
  });

  test('Very small numbers', () => ...);
  
  test('Very large numbers', () => ...);
})
```

Reply:
Ok, no need to complicate.